### DISCUSSION TITLE
toolbar search box "stuck" and always shows a search term when db is opened

### DISCUSSION THREAD

**Discussion by techiebiker1** [keepassxreboot/keepassxc#9900] [Q&A] [answered]
KeepassXC 2.7.5 is saving a search term and always populating the search field with that term when the database is opened.

The problem search term is "is:expired-3" without quotes.

There are no saved searches.

If the content of the search field is deleted, the "is:expired-3" term is deleted, and the database is closed then opened again, "is:expired-3" is again shown in the search field when the database is opened.

I found and set "Clear search query after" to 1 min. After applying this setting, if the database is opened and no action is taken for a minute the search box is cleared. At this point if the database is closed and then opened again, "is:expired-3" is again displayed in the search field when the database is opened.

If a new search term is entered then cleared and the database closed "is:expired-3" is again displayed in the search field when the database is next opened.

If a new search term is entered and the database closed while that new search term is displayed, "is:expired-3" is again displayed in the search field when the database is next opened.

How can this be resolved so the search field is empty when the database is opened?

**Comment by droidmonkey** [comment, marked as answer]
Go to application settings and disable the option to show expired entries on database unlock

**Reply by techiebiker1** [reply]
Thank you. It works.

I never would have made that connection and I can't find in the docs where that's described. I find the "Expired Entries" description but it makes no mention of that behavior or any indication that it applies to the search box.

I will submit an enhancement request because I can see no reason that "show entries" enabled should cause a search term that is cleared from the search box to reappear when the database is reopened.

**Reply by droidmonkey** [reply]
If you just lock the database it won't show again on unlock. If you close it then it will. 

https://keepassxc.org/docs/KeePassXC_UserGuide#_expired_entries

**Reply by michaelk83** [reply]
The reason you see the search term if "show expired entries" is enabled, is because that feature is implemented as a saved search for expired entries. If enabled, it simply loads the predefined saved search when you unlock the database for the first time. You can also manually load this search at any time from the sidebar, or enter that search term in the search box for the same results. The search term `is:expired-3` means "entries expiring within 3 days".

**Reply by techiebiker1** [reply]
When the setting "On database unlock, show entries that..." is enabled it causes the Search field to be populated when the database is opened. And the search term will always be of the form `is:expired-n`. I cannot find a "show expired entries" setting.

@droidmonkey provided the answer how to resolve the issue of the search field always being populated when the database is opened.

What I'm saying is the documentation does not give any indication that is what is going on. There's no common nomenclature in the docs or the UI to relate the two, "on database unlock show" and "search".

[Expired Entries](https://keepassxc.org/docs/KeePassXC_UserGuide#_expired_entries) indicates the default when the database is opened is to show entries that will expire in 3 days. It does not say enabling it will cause the Search field to be populated when the database is open and does not use the term "Search" when describing what Expired Entries does. And it says the feature can be changed or disabled in "Application Settings" but doesn't link to Application Settings. 

[Application Settings](https://keepassxc.org/docs/KeePassXC_UserGuide#_application_settings) doesn't refer to, illustrate, or explain "On database unlock..."

Not knowing why the Search field was populated every time I opened the database or how to prevent it is what started this thread.

I did/do read docs. At this point I would like to know how to make the documentation clear about the relationship so an always populated search field is explained for others who are confounded by that behavior. I would submit an edit(s) if I knew where/how.

**Reply by droidmonkey** [reply]
Edit the document here: https://github.com/keepassxreboot/keepassxc/blob/develop/docs/topics/DatabaseOperations.adoc#expired-entries

**Reply by michaelk83** [reply]
> I cannot find a "show expired entries" setting.

I may have gotten the setting name wrong.

As far as documentation, there should be a separate section on saved searches (a sub-section under [Searching the Database](https://keepassxc.org/docs/KeePassXC_UserGuide#_searching_the_database)). There's only a very brief mention of tags and pre-defined searches under [Application Layout](https://keepassxc.org/docs/KeePassXC_UserGuide#_application_layout), but it's very easy so miss, and there's no explanation what these pre-defined searches do, or how they relate to manually saved searches. And it only briefly alludes to the expired entries.

IMO, a big part of this confusion is that users are not familiar with the saved searches functionality.

**Comment by techiebiker1** [comment]
I tried.
No idea what I'm doing. Never used Github before.
I believe I've modified the "Expired Entries" section. 

Tired to modify the "Application Settings" settings section and add an image. Pretty sure that wasn't successful.

 

    On Friday, October 6, 2023 at 07:58:23 AM EDT, michaelk83 ***@***.***> wrote:  
 
 


The reason you see the search term if "show expired entries" is enabled, is because that feature is implemented as a saved search for expired entries. If enabled, it simply loads the predefined saved search when you unlock the database for the first time. You can also manually load this search at any time from the sidebar, or enter that search term in the search box for the same results. The search term is:expired-3 means "entries expiring in 3 days".

**Comment by techiebiker1** [comment]
Am trying to make modifications.
Never used Github before. Am LOST.
Believe have modified my "local repository" with changes to "Expired Entries".
Don't believe I've been able to modify my local repository with changes to "Application Settings" and not successful to add appropriate image.
Any SIMPLE task based videos you can direct me to for how to...
* add a link that references an existing page/section* add new images in local repository and "submit?" for inclusion

ty
 

    On Friday, October 6, 2023 at 04:29:34 PM EDT, michaelk83 ***@***.***> wrote:  
 
 



I cannot find a "show expired entries" setting.


I may have gotten the setting name wrong.

As far as documentation, there should be a separate section on saved searches (a sub-section under Searching the Database). There's only a very brief mention of tags and pre-defined searches under Application Layout, but it's very easy so miss, and there's no explanation what these pre-defined searches do, or how they relate to manually saved searches. IMO, a big part of this confusion is that users are not familiar with the saved searches functionality.
